### PR TITLE
Source generation for STJ update

### LINKED ISSUE #31413: Update source gen for .NET 7
There’s a new way to specify source generation mode in .NET 7, and that’s by specifying the new TypeInfoResolver property on JsonSerializerOptions.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: ee53c3dc-df79-82ef-ce25-3233ff292c55
* Version Independent ID: 558903e9-8173-464e-ff8b-7744419ad7e4
* Content: [How to use source generation in System.Text.Json](https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/source-generation)
* Content Source: [docs/standard/serialization/system-text-json/source-generation.md](https://github.com/dotnet/docs/blob/main/docs/standard/serialization/system-text-json/source-generation.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @gewarren
* Microsoft Alias: **gewarren**

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #31413.

### CODE REVIEW

File: docs/standard/serialization/system-text-json/source-generation-modes.md [hunk 1]
```diff
@@ -105,18 +105,22 @@ The following table shows which options in `JsonSerializerOptions` are supported
 | <xref:System.Text.Json.JsonSerializerOptions.NumberHandling>           | ❌                          |
 | <xref:System.Text.Json.JsonSerializerOptions.PropertyNamingPolicy>     | ✔️                         |
 | <xref:System.Text.Json.JsonSerializerOptions.ReferenceHandler>         | ❌                          |
+| <xref:System.Text.Json.JsonSerializerOptions.TypeInfoResolver>         |                             |
```
Comment (gewarren) [inline docs/standard/serialization/system-text-json/source-generation-modes.md:108]: I added some new options and attributes here and in the next table - if you can suggest if they're supported for source gen or not that would be great.

Comment (eiriktsarpalis) [reply docs/standard/serialization/system-text-json/source-generation-modes.md:108]: They're all supported in source gen 👍

**Review by eiriktsarpalis — APPROVED** [review]

**Review by BillWagner — APPROVED** [review]
This LGTM.

Do you want to ping @tdykstra for a look?

File: docs/standard/serialization/system-text-json/source-generation.md [hunk 2]
```diff
@@ -17,24 +16,19 @@ ms.topic: how-to
...
 ## Use source generation defaults
 
 To use source generation with all defaults (both modes, default options):
 
-* Create a partial class that derives from <xref:System.Text.Json.Serialization.JsonSerializerContext>.
-* Specify the type to serialize or deserialize by applying <xref:System.Text.Json.Serialization.JsonSerializableAttribute> to the context class.
-* Call a <xref:System.Text.Json.JsonSerializer> method that takes a <xref:System.Text.Json.Serialization.Metadata.JsonTypeInfo%601> instance. An alternative is to call a <xref:System.Text.Json.JsonSerializer> method that takes a <xref:System.Text.Json.Serialization.JsonSerializerContext> instance.
+1. Create a partial class that derives from <xref:System.Text.Json.Serialization.JsonSerializerContext>.
+1. Specify the type to serialize or deserialize by applying <xref:System.Text.Json.Serialization.JsonSerializableAttribute> to the context class.
+1. Call a <xref:System.Text.Json.JsonSerializer> method that either:
```
Comment (tdykstra) [inline docs/standard/serialization/system-text-json/source-generation.md:27]: "either" implies two options, but there are 3

**Review by tdykstra — APPROVED** [review]
LGTM, one comment

File: docs/standard/serialization/system-text-json/source-generation.md (hunk 2 above)
Comment (gewarren) [reply docs/standard/serialization/system-text-json/source-generation.md:27]: According to this, "either" is fine to introduce more than two phrases.

https://www.grammarphobia.com/blog/2017/08/either-neither.html

**Pull request merged**